### ISSUE TITLE
Multiple states

### ISSUE THREAD

**Issue by AngelaRT** [kenkeiras/TechTree#1]
It would be nice to have more that 2 states for the tasks. 

Currently, we have "TO-DO" and "Completed". 

For example, today I found that I needed something like "Discarted" to indicate that the task was supposed to be done, but it has been discarted. We don't delete it because we would like to keep a track of what we initially planned.

**Comment by kenkeiras** [comment]
Makes sense, maybe it would make sense to allow "Archiving" tasks (to keep common wording).

I'll look into this, thanks for the suggestion!

**Comment by AngelaRT** [comment]
I have just realized that we also would need a "_Doing_" state.

Right now I am reviewing my project tasks and I don't know how many of them are already pending and which ones are already started...

**Comment by kenkeiras** [comment]
Makes sense, I'm not sure about the final shape of this (in UX terms), but there's a rough aproximation deployed at the main server. (This corresponds to the branch `multiple-state-options` ). I'd be interested on hearing your opinion.

**Comment by kenkeiras** [comment]
I'm pretty happy with the current result, merging it to master as it won't hurt. 

Opinions are still welcome, I'll take another look at the possibilities when restyling the step cards.

**Comment by AngelaRT** [comment]
It looks much better now! I like it!
The selection of colours is great, since it is not too much distracting

**Comment by AngelaRT** [comment]
Btw, should you consider the dependencies regarding the states?

For example: A -> B

Where B depends on A completion. Then, if A is "Work in progress", and we change B to "Done", I think that at least a warning should pop up to let us know that it's weird that we finish a task whose dependency is not yet solved.

**Comment by kenkeiras** [comment]
Thanks!

What you say about warning the users when marking tasks as done makes a lot  of sense, I'll keep this open issue open then to remember that.
